File: .net & mono/Weed3Demo/demo_plus/queryProcedure/user_get2.cs
using Noear.Weed;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

//如果原来是存储过程的代码，可以通过[DbQueryProcedure]快速切换过来

namespace Weed3Demo.demo_plus.queryProcedure {
    class user_get2 : DbQueryProcedure {
        public user_get2() : base(DbConfig.test) {
            sql("select * from user where userID=@userID");

            set("@userID", () => userID);
        }

        public long userID;
    }
}

File: java/Weed3Demo/src/main/java/weed3demo/demo_plus/queryProcedure/user_get2.java
package weed3demo.demo_plus.queryProcedure;

import noear.weed.DbQueryProcedure;
import noear.weed.DbStoredProcedure;
import weed3demo.config.DbConfig;

/**
 * Created by noear on 2017/7/22.
 */

//如果原来是存储过程的代码，可以通过[DbQueryProcedure]快速切换过来

public class user_get2 extends DbQueryProcedure {
    public user_get2() {
        super(DbConfig.test);

        sql("select * from user where userID=@userID");
        set("@userID", () -> userID);
    }

    public long userID;
}
